src/compiler/utilities.rs

```diff
@@ -476,24 +493,17 @@ impl DiagnosticCollection {
     }
 
     pub fn add(&mut self, diagnostic: Rc<Diagnostic>) {
-        // let diagnostics = self
-        //     .file_diagnostics
-        //     .get_mut(&diagnostic.file().unwrap().file_name)
-        //     .unwrap_or_else(|| {
-        //         let diagnostics: SortedArray<Rc<Diagnostic>> = SortedArray::new(vec![]);
-        //         self.file_diagnostics.insert(
-        //             diagnostic.file().unwrap().file_name.to_string(),
-        //             diagnostics,
-        //         );
-        //         self.file_diagnostics
-        //             .get_mut(&diagnostic.file().unwrap().file_name)
-        //             .unwrap()
-        //     });
         if let Some(diagnostics) = self
             .file_diagnostics
             .get_mut(&diagnostic.file().unwrap().file_name)
         {
-            insert_sorted(diagnostics, diagnostic);
+            insert_sorted(
+                diagnostics,
+                diagnostic,
+                |rc_diagnostic_a: &Rc<Diagnostic>, rc_diagnostic_b: &Rc<Diagnostic>| {
```

Comment:
I guess naming this `rc_*` became a little unnecessary after it complained that it needed explicit type annotations here